src/pages/Comments.jsx

```diff
@@ -1,10 +1,22 @@
 import CommentCard from "../cards/CommentCard";
-import DeleteCommentBtn from "../components/DeleteCommentBtn";
-import { useContext } from "react";
+import DeleteButton from "../styleComponents/DeleteButton";
+import { useContext, useState } from "react";
 import { UserContext } from "../contexts/UserContext";
+import { deleteCommentById } from "../../utils/api";
 
-export default function Comments({ comments }) {
+function Comments({ comments, setComments }) {
   const { user } = useContext(UserContext);
+  const [isDeleted, setIsDeleted] = useState(false);
+  function handleDelete(id, e) {
+    e.preventDefault();
+    deleteCommentById(id);
+    const newComments = comments.filter((comment) => {
+      return comment.comment_id !== id;
+    });
```

Comment:
If you do the filter before the deleteCommentById (literally just move them around) then this would be performing optimistically and you aren't waiting for the delete to work on the API to change the state